src/hooks/useChannel.test.tsx

```diff
@@ -143,6 +142,63 @@ describe('useChannel', () => {
         expect(onConnectionError).toHaveBeenCalledTimes(1);
         expect(onConnectionError).toHaveBeenCalledWith(reason);
     });
+
+    it('should use the latest version of the message callback', async () => {
+        let callbackCount = 0;
+
+        const TestComponent = () => {
+            const [count, setCount] = React.useState(0);
+
+            useChannel('blah', () => {
+                callbackCount++;
+                setCount((prev) => prev + 1);
+            });
+
+            return <div role="counter">{count}</div>;
+        };
+
+        renderInCtxProvider(ablyClient, <TestComponent />);
+
+        await act(async () => {
+            ablyClient.channels.get('blah').publish({ text: 'test message 1' });
```

Comment:
I think we need to publish in different acts, to ensure state has been updated

Reply:
Done 😄